### DISCUSSION TITLE
Custom filter variable & Permissions

### DISCUSSION THREAD

**Discussion by Podolchak** [directus/directus#19512] [Feature Requests]
### Summary

I'd like to see a mechanism that would allow you to specify your own parameters when reading collections and produce results based on those parameters. For example, like $CURRENT_USER, but with a custom parameter like $CUSTOM. Thus we get the ability to produce results only if we passed the correct parameter for which the results will be found, otherwise we will get an empty object.

### Motivation

Previously, before the Directus websocket, I used to get around this by creating my own endpoint, but now that you can "stick" to a collection and listen to it with a websocket, this feature makes sense.

### Detailed Design

I think it doesn't require any further explanation )

### Requirements List

This little innovation, I believe, will help unlock the potential of websocket even further

### Drawbacks

I think this feature should be added )

### Alternatives

-

### Adoption Strategy

-

**Comment by rijkvanzanten** [comment]
> I think it doesn't require any further explanation )

I ehh.. disagree. I don't really understand what you mean! Could you share an example of how you would use this in practice?

**Reply by Podolchak** [reply]
<img width="836" alt="" src="https://github.com/directus/directus/assets/141161694/a1d5d5b7-2e0d-44fe-8124-92dcc350e620">

Allow to pass a custom parameter to the body request

**Reply by Podolchak** [reply]
For example, there is a certain reference
If its slug matches a record in the collection, it returns data that has the specified slug in my parameter

Or for example there is a slug of a restaurant, and accordingly we pull up the dishes of this restaurant

And if this slug, or for example some UUID is not in the body of the request - we just return an empty object.

So we do not throw all the data in the public and we have the opportunity to give unique content through "secret" links, only to those who have this unique URL for example

Accordingly, thanks to this we will be able to listen to the collection through the socket receiving updates that meet the criteria, and without this we will have to throw all the data into the full public and filter for ourselves or add authorization

**Reply by Podolchak** [reply]
In general, this will be a small mechanism that can restrict data retrieval beyond authorization

**Reply by rijkvanzanten** [reply]
How would you set the value of that custom flag? How does it differ from just regular filtering?

**Reply by Podolchak** [reply]
It differs in the fact that we specify what criterion records should meet, and this criterion can even be used in any way, even organize access to the profile by link

Just pass it in the body of a read or create request, e.g.

```js
readItems('collection', {
      body: {
        slug: 123,
      },
    }),
```
    
Filtering is good, but in this case, we immediately give access to a person to all records, which can simply open the source and execute a modification request to get all the data at once.

If we want to protect the data, then we only have authorization, and we can, for example, organize data access to ensure that only the person who has the link has access to the data.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
